src/v3/src/util/settingsUtils.ts

```diff
@@ -246,3 +248,32 @@ export const isOauth2Enabled = (widgetProps: WidgetProps): boolean => {
   }
   return typeof clientId !== 'undefined' && authScheme?.toLowerCase() === 'oauth2';
 };
+
+export const getPageTitle = (
+  widgetProps: WidgetProps,
+  formTitle: string,
+  idxTransaction?: IdxTransaction,
+): string | undefined => {
+  const { brandName, features: { setPageTitle } = {} } = widgetProps;
+
+  // When setPageTitle option is 'undefined', default will be to set title based on page header
+  // When setPageTitle option is 'true', set title based on page header
+  if ((typeof setPageTitle === 'boolean' && setPageTitle === true)
+    || typeof setPageTitle === 'undefined') {
+    return brandName ? `${brandName} | ${formTitle}` : formTitle;
+  }
+
+  if (typeof setPageTitle === 'string') {
+    return setPageTitle;
+  }
+
+  if (typeof setPageTitle === 'function') {
+    const eventContext: EventContext = typeof idxTransaction === 'undefined'
+      ? { controller: null }
+      : getEventContext(idxTransaction);
+    return setPageTitle(eventContext, { formTitle, brandName });
+  }
+
+  // Indicates setPageTitle config option was purposefully disabled
+  return undefined;
```

Comment:
nit: return `null` rather than `undefined` when an empty return value is intentional